ocrd_tesserocr/segment_region.py

```diff
@@ -1,38 +1,33 @@
 from __future__ import absolute_import
 
 from ocrd_utils import getLogger
-from ocrd import Processor
+from ocrd_validators import ParameterValidator
 
 from .config import OCRD_TOOL
 from .recognize import TesserocrRecognize
 
 TOOL = 'ocrd-tesserocr-segment-region'
+BASE_TOOL = 'ocrd-tesserocr-recognize'
 
-class TesserocrSegmentRegion(Processor):
-
+class TesserocrSegmentRegion(TesserocrRecognize):
     def __init__(self, *args, **kwargs):
-        kwargs['ocrd_tool'] = OCRD_TOOL['tools'][TOOL]
-        kwargs['version'] = OCRD_TOOL['version']
+        kwargs.setdefault('ocrd_tool', OCRD_TOOL['tools'][TOOL])
+        # workaround for core#998
+        if kwargs['ocrd_tool'] is None:
+            kwargs['ocrd_tool'] = OCRD_TOOL['tools'][TOOL]
         super().__init__(*args, **kwargs)
-        
-        if hasattr(self, 'workspace'):
```

Comment:
Because there are different contexts for processor instances (depending on CLI arguments):
- instances that will be passed to `run_processor` (which ultimately runs `process`, ideally repeatedly)
- instances that exit immediately and print/dump some info

To differentiate this, by the CLI and API we can only check whether or not an `output_file_grp` (or `parameter` or `workspace`) kwarg exists.